### PR TITLE
feat(SPM-1306): create a patch set wizard

### PR DESCRIPTION
**Pull request by mkholjuraev**
We have incomplete UI mockup designs. I would like to improve the wizard once the mockups are complete.

### CODE REVIEW

File: src/SmartComponents/PatchSetWizard/PatchSetWizard.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                                        state={wizardState}
+                                    />
+                                ),
+                                isFinishedStep: true
+                            }
+                        ]}
+                        onClose={handleWizardClose}
+                    />
+                </Modal>
+            ) : null}
```
Comment (bastilian) [inline src/SmartComponents/PatchSetWizard/PatchSetWizard.js:121]: Is there a specific reason why this is not just one `{ ... ? ... : ... }` ?

File: src/SmartComponents/PatchSetWizard/steps/RequestProgress.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                    : percent === 100
+                        ? 'Patch Set creation successfull'
+                        : 'Patch Set creation in progress'}
+            </Title>
+            <EmptyStateBody>
+                <Grid hasGutter>
+                    <GridItem>
+                        <ProgressBar percent={percent} failed={failed} />
+                    </GridItem>
+                    {(percent !== 100 && !failed) && (
```
Comment (bastilian) [inline src/SmartComponents/PatchSetWizard/steps/RequestProgress.js:58]: Could you not just wrap the second `GridItem` with the condition(s)?

File: src/SmartComponents/PatchSetWizard/InputFields/DescriptionField.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React, { useState } from 'react';
+import { FormGroup, TextInput } from '@patternfly/react-core';
+import useFormApi from '@data-driven-forms/react-form-renderer/use-form-api';
+import useFieldApi from '@data-driven-forms/react-form-renderer/use-field-api';
+
+const DescriptionField = (props) => {
+    const { input } = useFieldApi(props);
+    const formOptions = useFormApi();
+    const values = formOptions.getState()?.values;
```
Comment (bastilian) [inline src/SmartComponents/PatchSetWizard/InputFields/DescriptionField.js:9]: Is it not possible to use `props.value(s)`, like in the second example here: https://data-driven-forms.org/hooks/use-field-api#usageofinput

File: src/store/Reducers/PatchSetsReducer.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import * as ActionTypes from '../ActionTypes';
+
+// Initial State. It should not include page and perPage to persist them dynamically
```
Comment (bastilian) [inline src/store/Reducers/PatchSetsReducer.js:3]: > It should not include page and perPage to persist them dynamically

... ähm... but it does include page and perPage? :D

**Review by bastilian — APPROVED** [review]
Looks good to me! And works as expected! Thank you @mkholjuraev!

**Comment by mkholjuraev** [comment]
@bastilian Thanks for the review. I will adopt your suggestions and merge the PR

File: src/SmartComponents/PatchSetWizard/steps/RequestProgress.js (hunk 2 above)
Comment (mkholjuraev) [reply src/SmartComponents/PatchSetWizard/steps/RequestProgress.js:58]: We should display different buttons also. Thus I think it makes sense to have a separate GridItem

Comment (mkholjuraev) [reply src/SmartComponents/PatchSetWizard/steps/RequestProgress.js:58]: Anyways, these lines will be most probably changed when the designs are ready for the last step.

File: src/SmartComponents/PatchSetWizard/InputFields/DescriptionField.js (hunk 3 above)
Comment (mkholjuraev) [reply src/SmartComponents/PatchSetWizard/InputFields/DescriptionField.js:9]: I am not sure if the props would always have the value when user comes back to the input from next step.

console.log(props) does not list the value in that ^ senario.

image

**Pull request merged**